File: solutions/java/src/trafficsignalsystem/YellowState.java
package trafficsignalsystem;

public class YellowState implements SignalState {
    @Override
    public void handle(TrafficLight light, TrafficSignalController controller, Direction direction) {
        System.out.println("Direction: " + direction + " | State: YELLOW");
        int duration = controller.getSignalDuration(direction, this);
        controller.scheduleStateChange(light, direction, new RedState(), duration);
    }
    @Override
    public String getName() { return "YELLOW"; }
} 

File: solutions/typescript/src/TrafficSignalSystem/SignalState/YellowState.ts
import { DirectionEnum } from "../Direction";
import TrafficLight from "../TrafficLight";
import TrafficSignalController from "../TrafficSignalController";
import RedState from "./RedState";
import SignalState from "./SignalState";

export default class YellowState implements SignalState {
    handle(
        light: TrafficLight,
        controller: TrafficSignalController,
        direction: DirectionEnum,
    ) {
        console.log("Direction: " + direction + " | State: YELLOW");
        const duration = controller.getSignalDuration(direction, this);
        controller.scheduleStateChange(
            light,
            direction,
            new RedState(),
            duration,
        );
    }

    getName() {
        return "YELLOW";
    }
}
